solana/solana-ibc/programs/solana-ibc/src/trie_key.rs

```diff
@@ -232,6 +231,14 @@ impl AsComponent for ibc::core::ics24_host::identifier::ChannelId {
     }
 }
 
+impl AsComponent for ibc::Height {
+    fn key_len(&self) -> usize { 2 * 0_u64.key_len() }
+    fn append_into(&self, dest: &mut Vec<u8>) {
```

Comment:
why not dest: &mut [u8] or `AsMut<[u8]>`?

Reply:
Vec reallocates and tracks length of written data.  With `&mut [u8]` will have to preallocate maximum possible length and then extract length at the end.  Having said that, I will change it to `&mut &[u8]` once I convert all identifiers to integers.  Then we’ll be able to use ~30 byte fixed-size buffer for trie key.

Reply:
I’ve forgotten that we’re already calculating the length of the key and are doing proper preallocation, so yeah we could be using `&mut &[u8]` here even now.  Regardless, this is still planned once I convert all identifiers to integers.